staging/src/k8s.io/component-base/metrics/prometheus/restclient/metrics.go

```diff
@@ -99,15 +108,14 @@ var (
 )
 
 func init() {
-	execPluginCertTTL.Set(math.Inf(1)) // Initialize TTL to +INF
 
 	legacyregistry.MustRegister(requestLatency)
 	legacyregistry.MustRegister(deprecatedRequestLatency)
 	legacyregistry.MustRegister(requestResult)
-	legacyregistry.MustRegister(execPluginCertTTL)
+	legacyregistry.RawMustRegister(execPluginCertTTL)
 	legacyregistry.MustRegister(execPluginCertRotation)
 	metrics.Register(metrics.RegisterOpts{
-		ClientCertTTL:         &ttlAdapter{m: execPluginCertTTL},
```

Comment:
you can accomplish this without changing the client-go API by taking the ttl passed in, converting it to a time, and then converting back to a duration  when read

Reply:
I am not sure if I am understanding your suggestion correctly but if I am it sounds kind of gross to me.

So you're saying change it to:

time(cert expr) -> duration(expr - now) -> time(ClientCertTTL metric) -> duration(TTL - now)?

Theres no reason to preserve the client-go API since I was the one who defined this part of it in my last CL. Its brand new and now is the best time to change it.

Reply:
> Theres no reason to preserve the client-go API since I was the one who defined this part of it in my last CL. Its brand new and now is the best time to change it.

It's in 1.17, right?

Reply:
nope, I was wrong

https://github.com/kubernetes/kubernetes/blob/release-1.17/staging/src/k8s.io/client-go/tools/metrics/metrics.go
vs
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/metrics/metrics.go